### DISCUSSION TITLE
ban / blacklist misbehaved users

### DISCUSSION THREAD

**Discussion by yeahletters** [jamulussoftware/jamulus#3030] [Ideas]
Is there a function to ban or blacklist certain users with unique IP addresses on my own jamulus server? I had an unwanted guest (troll) posting profanties in the chat using capitals.

**Comment by ann0see** [comment]
Hi @yeahletters That's bad to hear.

 Yes, there are unofficial ones like https://github.com/dingodoppelt/jamban and @Rob-NY also has one. 

We have multiple requests like https://github.com/jamulussoftware/jamulus/discussions/1057 or https://github.com/jamulussoftware/jamulus/discussions/1371 but I think there's a general disagreement about this topic I'm afraid.

**Comment by ann0see** [comment]
Note: I converted this to a discussion as it is not an actionable issue. You can now find your issue in the forum: https://github.com/jamulussoftware/jamulus/discussions/3030

**Comment by ann0see** [comment]
Since we had quite some discussion about this topic, I belive we should really consider implementing something...

**Reply by Rob-NY** [reply]
@ann0see - When pr #2918 is merged I'm happy to introduce some other dependency PR's that offer an RPC-based approach to this issue as well as some others.  Unfortunately, all of them (like this one) are somewhat controversial.

**Reply by ann0see** [reply]
I've just had a look at #2918 again. There might be some smaller things but it's close.

**Comment by mcfnord** [comment]
I wonder if they're using a VPN to get unique IP addresses. Perhaps you could ban all IP ranges of that VPN. There aren't good reasons to use a VPN with Jamulus, are there? Perhaps only bad reasons.

**Comment by yeahletters** [comment]
Had someone post profanities in the chat again tonight. This time round it was this:*** (removed profanities)

**Reply by mcfnord** [reply]
Try blocking their IP at the firewall.

**Reply by yeahletters** [reply]
I cannot do this if the abuse was not on my own server

**Reply by ann0see** [reply]
I still believe that the server owner should be the one responsible for general banning. In your case: did you contact the server owner?

**Reply by yeahletters** [reply]
No, I don't know who the owner is of the server, I have since checked the chat log and I can't find this information anywhere.

**Reply by iknowkungfutoo** [reply]
@ann0see How would you know who the server owner is or what their contact details are?

**Comment by yeahletters** [comment]
It has happened again guys, I need to have a moderated chat please, this is happening to me too often. I don't feel safe or secure anymore. Maybe it is true that I need muting, but I really objectively speaking do not think so. (12:50:32 AM) No nom She need mute

**Reply by mcfnord** [reply]
Maybe you need the power to turn off chats from everyone for a duration that defaults to an hour.

**Comment by yeahletters** [comment]
This would mean that I would miss chats from people who really have something to say, not ideal.

**Comment by yeahletters** [comment]
I will paste the chat history (which happened last night), which I have coincidently been saving as a private logbook since the uncertainty of this not currently yet being a feature.

To protect my own sanity I have not read the chat because it is hurtful and damaging.

As mcfnord has mentioned banning on a firewall level is only possible for me on my own server. On this occasion I was not on my own server but it happened on the 'De Insel 2.0' server on 30 March 2026. I am afraid that I am compelled in conclusion to add that these kind of episodes have happened as a regular occurence. Please add a moderator chat function in future releases.   

(09:27:59 PM) Hoi-yin 🎙️🎶 https://drive.google.com/drive/folders/1btYE0NTgsR0AuL6WeWF7GJ09oc2DsMtu
(09:35:36 PM) Hoi-yin 🎙️🎶 baum
(09:43:42 PM) Hoi-yin 🎙️🎶 https://tabs.ultimate-guitar.com/tab/kandace-springs/novocaine-heart-chords-6198377
(09:50:52 PM) Hoi-yin 🎙️🎶 https://tabs.ultimate-guitar.com/tab/kandace-springs/place-to-hide-chords-3024797
(09:52:18 PM) NJoy uhate it
(09:52:20 PM) NJoy but love it
(09:53:34 PM) NJoy living u to my street cred
(09:53:36 PM) NJoy what u got?
(09:55:19 PM) NJoy ffs vst new to this convo?
(09:55:24 PM) NJoy wtaf!!!
(09:55:35 PM) Cameron hi
(09:56:02 PM) NJoy vst 2 yeah
(09:56:04 PM) NJoy check em
(09:56:06 PM) NJoy obsolete
(09:56:29 PM) NJoy yep only coz never been honourale from her
(09:56:31 PM) NJoy lol
(09:56:34 PM) NJoy .com
(09:56:37 PM) NJoy avit
(09:57:00 PM) NJoy love mn ffs
(09:57:04 PM) NJoy for fuck sake
(09:57:50 PM) Hoi-yin 🎙️🎶 what the fucking hell Alan?
(09:58:00 PM) Hoi-yin 🎙️🎶 fuck the shit off
(09:58:21 PM) NJoy country?
(09:58:23 PM) NJoy makes sense
(10:01:24 PM) NJoy yep, same as 4 yrs ago
(10:01:27 PM) NJoy still nice
(10:01:30 PM) NJoy ust same
(10:01:37 PM) NJoy very nice
(10:01:39 PM) NJoy come on
(10:01:46 PM) NJoy but itch correct required imo
(10:01:54 PM) NJoy or think more
(10:03:01 PM) NJoy haha, same old Hoi nothing hoi
(10:03:03 PM) NJoy wow
(10:03:07 PM) NJoy been compationate mate
(10:03:10 PM) NJoy you eny shit
(10:03:19 PM) NJoy honestly i mean it nicely
(10:03:22 PM) NJoy i used to think
(10:03:24 PM) NJoy all sorts
(10:03:27 PM) NJoy now i listen
(10:04:07 PM) NJoy si
(10:04:10 PM) NJoy same ol'd
(10:04:16 PM) NJoy be fucking nice
(10:04:25 PM) NJoy unless you got a reason
(10:04:42 PM) NJoy love hoi
(10:04:50 PM) NJoy im not being nasty just honest
(10:04:54 PM) NJoy im a cock audio wise
(10:04:59 PM) NJoy your personality suckstho
(10:05:37 PM) Hoi-yin 🎙️🎶 +5
(10:05:42 PM) Hoi-yin 🎙️🎶 https://tabs.ultimate-guitar.com/tab/2948705
(10:08:43 PM) 𝄢Tom hello all
(10:10:19 PM) Rinaldo🕸️ 🎶 https://tabs.ultimate-guitar.com/tab/2948705
(10:11:42 PM) Hoi-yin 🎙️🎶 https://tabs.ultimate-guitar.com/tab/debarge/all-this-love-chords-5989154
(10:13:59 PM) Rudi M hi
(10:19:34 PM) 𝄢Tom Hello Rudi,
(10:19:56 PM) 𝄢Tom alternate
(10:20:21 PM) 𝄢Tom you play the next
(10:20:21 PM) Rudi M no thx, to late for me
(10:20:30 PM) 𝄢Tom ok
(10:20:34 PM) Rudi M thx again
(10:24:16 PM) Hoi-yin 🎙️🎶 https://tabs.ultimate-guitar.com/tab/wet-wet-wet/angel-eyes-chords-3096671
(10:30:44 PM) Hoi-yin 🎙️🎶 Hola Angel
(10:32:06 PM) Angel hi
(10:32:33 PM) Hoi-yin 🎙️🎶 https://tabs.ultimate-guitar.com/tab/johnny-hates-jazz/turn-back-the-clock-chords-5099137
(10:32:36 PM) Rens https://tabs.ultimate-guitar.com/tab/johnny-hates-jazz/turn-back-the-clock-chords-5099137
(10:36:42 PM) Rudi M good night cu

**Reply by ann0see** [reply]
Would it help if you could block the user from the client side such that you don't get messages/hear them?

As we don't have unique user names/accounts/... users could just change their name/ip/... to circumvent any block.

**Reply by yeahletters** [reply]
I can't do this. How does one 'block' certain users on a public server?

**Reply by ann0see** [reply]
No. It's just an idea for a feature

**Reply by yeahletters** [reply]
Thanks Ann. There's unfortunately no way to circumvent one not using their own (real) name, or using a nickname either, or not using a name at all (blank).

**Reply by mcfnord** [reply]
I think it's naive the way Windows draws attention to a new chat. On Windows, I put the chat window on a different desktop. (Virtual desktops are a feature of Windows 11.) This lets a harasser rage all day and I don't see anything. But why is cleverness required to accomplish this?

I support a simple, practical feature: if someone's audio is blocked by enough people, also quietly discard their chats without delivering them. This gives the mob some power. Someone that drunk or sad should just rage on and be invisible. This change would only work for dumb drunks, but dumb drunks are the bulk of these scenarios.

**Comment by iknowkungfutoo** [comment]
I was just thinking about this a couple of evenings ago. Add a mute chat button below the mute button for each channel. If pressed, any char from that channel is not shown on the chat window.

**Reply by ann0see** [reply]
I'd prefer something unobtrusive: right click on mute -> context me u saying Block and Mute, Report to server owner, ...

Chat blocking will either need client side omission of displaying text messages from a user (bad design bud backwards compatible 
) or server side blocking where the server just doesn't send the message anymore (better design, not backwards compatible)

**Reply by iknowkungfutoo** [reply]
Why is it a bad design? It follows the K.I.S.S. mantra and should be straight forward to implement. On the other hand, if you think it should be done server side, that means passing messages back to the server and the server then with holding chat messages from a user to a specific user. It also requires server owners to update. So, from your perspective, as a user, you could well still suffer the problem that you are trying to solve. But if it is done client side, you can install a client and then have the tools to mute audio and chat on any server that you join.  With that all said, the simple client-side solution is a better solution and design (as it is simple).

**Reply by ann0see** [reply]
The client does not know who sent the message. It just prints everything the server sends. 

Therefore the client would need to do some regex magic to filter out messages from a specific user. This may break if there's a message containing a string on which the regex matches but from another user. 
Thus client side blocking is fragile. 

Concerning adding a block button: I want to have a simple UI. Users should not have 4 buttons below the fader if there's block button that is only used sometimes or a one time thing.

**Comment by yeahletters** [comment]
Had someone post an unnecessary and undesirable word again yesterday on my own server. I was too late to note the IP address of that particular user, but made a mental note of the username.  

(01:01:09 AM) wolle schiß
(01:04:56 AM) Stéphane https://fr.ultimate-guitar.com/tab/talk-talk/such-a-shame-chords-1190742
(01:08:49 AM) Rudi M it's bed time for me. good night guys

**Reply by ann0see** [reply]
The word with the "ß"?

**Comment by yeahletters** [comment]
Yes

On Sun, Apr 26, 2026 at 8:52 AM ann0see ***@***.***> wrote:

> The word with the "ß"?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/jamulussoftware/discussions/3030#discussioncomment-16716614>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH7K5FKFVUZWHNPRQLWMXVD4XWW4HAVCNFSM6AAAAACXHH32A6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTMNZRGY3DCNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
hoiyin ip

**Reply by ann0see** [reply]
It's misspelled if it's German and not extremely bad. Just so you know.

**Reply by ann0see** [reply]
But obviously it depends on the context. I'd probably just say something like "mind your language and stay polite".

**Comment by yeahletters** [comment]
I see, yes German. Not so bad, but unnecessary, and again a case of two
seconds in the server, leaving the rude comment and then prompt departure.
So bad manners and unnecessary voicing of one's not appreciated opinion.

On Sun, Apr 26, 2026 at 1:49 PM ann0see ***@***.***> wrote:

> But obviously it depends on the context. I'd probably just say something
> like "mind your language and stay polite".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/jamulussoftware/discussions/3030#discussioncomment-16718211>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH7K5FLOL6B46TWNLSTPDRT4XXZV5AVCNFSM6AAAAACXHH32A6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTMNZRHAZDCMI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
hoiyin ip
